src/utils/hasPropsDeepEqual.js

```diff
@@ -0,0 +1,25 @@
+import xorWithLodash from 'lodash/xorWith'
+import isEqualLodash from 'lodash/isEqual'
+import isEmptyLodash from 'lodash/isEmpty'
+
+/**
+ * Compare if the new props are equal or not to the old props
+ *
+ * @param {object} prevProps - Previous props
+ * @param {object} nextProps - Next Props
+ * @returns {boolean} - If prevProps & nextProps are equals
+ */
+export const hasPropsDeepEqual = (prevProps, nextProps) => {
+  const prevPropsArr = Object.keys(prevProps)
+  const nextPropsArr = Object.keys(nextProps)
+
+  if (prevPropsArr.length !== nextPropsArr.length) return false
+
+  const res = prevPropsArr.map(key => {
+    return isEmptyLodash(
+      xorWithLodash(prevProps[key], nextProps[key], isEqualLodash)
+    )
+  })
```

Comment:
Let's use some in order to exit as soon as possible. 

But we should not rely on this kind of function. It's expensive, and sometimes having a re-render with no change is faster...

Reply:
Solved above